src/nvim/lua/executor.c

```diff
@@ -2217,3 +2218,46 @@ char *nlua_read_secure(const char *path)
 
   return buf;
 }
+
+void ex_trust(exarg_T *eap)
+{
+  char *mode;
+  char *path;
+  {
+    char *p = skiptowhite(eap->arg);
+    size_t len = (size_t)(p - eap->arg);
+    mode = xcalloc(len + 1, sizeof(char));
+    memcpy(mode, eap->arg, len);
+    path = skipwhite(p);
+  }
+
+  {
+    lua_State *const lstate = global_lstate;
+    lua_getglobal(lstate, "vim");
+    lua_getfield(lstate, -1, "secure");
+    lua_getfield(lstate, -1, "trust");
+    lua_pushstring(lstate, path);
+    lua_pushstring(lstate, mode);
+    if (nlua_pcall(lstate, 2, 2)) {
+      nlua_error(lstate, _("Error executing vim.secure.trust: %.*s"));
+      return;
+    }
+
+    size_t len = 0;
+    const char *contents = lua_tolstring(lstate, -1, &len);
+    if (contents != NULL) {
+      // Add one to include trailing null byte
+      char *errmsg = xcalloc(len + 1, sizeof(char));
+      memcpy(errmsg, contents, len + 1);
+      if (errmsg != NULL) {
```

Comment:
Null check here is not necessary, `xcalloc` guarantees the returned pointer is valid.